build-fail-reminder.py

```diff
@@ -157,37 +188,40 @@ def main(args):
                                 'recepient': to,
                                 'sender': sender,
                                 'project' : project,
                                 'package' : package,
                                 'date': time.ctime(Reminded[package].firstfail),
                                 }
-                    msg = MIMEText(text, _charset='UTF-8')
-                    msg['Subject'] = '%s - %s - Build fail notification' % (project, package)
-                    msg['To'] = email.utils.formataddr((to, fullname))
-                    msg['From'] = sender
-                    msg['Date'] = email.utils.formatdate()
-                    msg['Message-ID'] = email.utils.make_msgid()
-                    msg.add_header('Precedence', 'bulk')
-                    msg.add_header('X-Mailer', '%s - Failure Notification' % project)
-                    logger.info("%s: %s", msg['To'], msg['Subject'])
-                    if args.dry:
-                        logger.debug(msg.as_string())
-                    else:
-                        try:
-                            s = smtplib.SMTP(args.relay)
-                            s.sendmail(msg['From'], {msg['To'], sender }, msg.as_string())
-                            s.quit()
-                        except:
-                            logger.error("Failed to send an email to %s (%s)" % (fullname, to))
-                            pass
-
-            elif Reminded[package].remindCount == 3:
-                logger.warning( "Package '%s' has been failing for three weeks - let's create a bug report" % package)
-            else:
-                logger.warning( "Package '%s' is no longer maintained - send a mail to factory maintainers..." % package)
+                    SendMail(logger, project, sender, to, fullname, subject, text)
+            elif Reminded[package].remindCount == 4:
+                # Package has failed for 4 weeks - Collect packages to send a mail to openSUSE-factory@ (one mail per day max)
+                ProjectComplainList.append(package)
+            elif Reminded[package].remindCount == 6:
+                # Package failed to build for 6 weeks - file a delete request
+                r = osc.core.Request()
+                r.add_action('delete', tgt_project=project, tgt_package=package)
+                r.description = cgi.escape("[botdel] Package has failed to build for >= 6 weeks")
+                r.create(apiurl)
+
+    if (len(ProjectComplainList)):
+        # At least to report to the project for not building - send a mail to openSUSE-Factory
+        ProjectComplainList.sort()
+        to = 'openSUSE-Factory@opensuse.org'
+        fullname = "openSUSE Factory - Mailing List"
+        subject = "%(project)s - Build fail notification" % {'project': project}
+        text = MAIL_TEMPLATES[2] % {
+            'project': project,
+            'packagelist': ProjectComplainList,
+            'sender': sender,
+            }
```

Comment:
packagelist and sender are not used in MAIL_TEMPLATES[2]

Reply:
Dude, this is your code :)

Reply:
I know - and this was from before I added the for pkg in blabla to list them nicely and not just in a weird way.

Reply:
Having this in the same array is strange anyway. I'll extract it